### PR TITLE
TRACING-3258 | Document troubleshooting OTEL

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):

4.11
4.12
4.13
4.14

Issue:

https://issues.redhat.com/browse/TRACING-3258

Link to docs preview:

https://64772--docspreview.netlify.app/

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by pavolloffay** [comment]
This can be merged ASAP, it is already present in RHOSDT 2.9

File: _topic_maps/_topic_map.yml
Comment (max-cx) [inline _topic_maps/_topic_map.yml:3668]: Apologies for the confusing. `Use` can be both a noun and a verb: on the previous line, it is a noun (so it's parallel to Migration and Removal).
```suggestion
    - Name: Troubleshooting
```

File: modules/distr-tracing-otel-troubleshoot.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot.adoc:45]:
```suggestion
<1> The address on which internal collector metrics are exposed. The operator creates a  `<cr-name>-collector-monitoring` service that you can use to scrape the metrics endpoint. Defaults to `:8888`.
```

File: modules/distr-tracing-otel-troubleshoot.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot.adoc:49]:
```suggestion
The logging exporter is configurable for exporting the collected data to the standard output.
```

File: distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting.adoc
Comment (max-cx) [inline distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting.adoc:11]:
```suggestion
include::modules/distr-tracing-otel-troubleshoot-collector-logs.adoc[leveloffset=+1]

include::modules/distr-tracing-otel-troubleshoot-metrics.adoc[leveloffset=+1]

include::modules/distr-tracing-otel-troubleshoot-logging-exporter-stdout.adoc[leveloffset=+1]
```

File: modules/distr-tracing-otel-troubleshoot.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot.adoc:1]: I recommend [removing this file](https://github.com/openshift/openshift-docs/pull/64772/files#r1331727890). It's failing the current build.

File: modules/distr-tracing-otel-troubleshoot-logs.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot-logs.adoc:9]: Found a misrendering in the preview:
```suggestion

.Procedure
```

File: distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting.adoc
Comment (max-cx) [inline distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting.adoc:9]:
```suggestion
The OpenTelemetry Collector offers multiple ways to measure its health as well as investigate data ingestion issues.
```

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:8]:
```suggestion
The OpenTelemetry Collector exposes the metrics about the data volumes it has processed. The following metrics are for spans, although similar metrics are exposed for metrics and logs signals:
```

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:13]:
```suggestion
`otelcol_receiver_accepted_spans`:: The number of spans successfully pushed into the pipeline.
`otelcol_receiver_refused_spans`:: The number of spans that could not be pushed into the pipeline.
`otelcol_exporter_sent_spans`:: The number of spans successfully sent to destination.
`otelcol_exporter_enqueue_failed_spans`:: The number of spans failed to be added to the sending queue.
```

File: modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc
Comment (max-cx) [inline modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc:30]:
```suggestion
. Use the `oc logs` command or the OpenShift console to export the logs to the standard output.
```

**Review by IshwarKanse — APPROVED** [review]

**Comment by max-cx** [comment]
Preview of the page created in this PR: https://64772--docspreview.netlify.app/openshift-enterprise/latest/distr_tracing/distr_tracing_otel/distr-tracing-otel-troubleshooting

**Comment by max-cx** [comment]
Peer/language review is done.

**Review by max-cx — APPROVED** [review]

File: modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc:6]: Nit: I think this would benefit from a more descriptive title, such as "Exporting collected data using the logging exporter".

File: modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc:30]: Nit: I think it would benefit customers to include substeps that show the `oc logs` command in a command block and another substep to show how to access logs in the console.

File: modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-logging-exporter.adoc:12]:
```suggestion
. Configure the OpenTelemetry Collector custom resource (CR) as follows:
```

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:19]:
```suggestion
. Enable the telemetry service by adding the following lines in the OpenTelemetry Collector custom resource (CR):
```

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:40]: An `.Example output` block would be a nice touch to help orient customers.

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:37]:
```suggestion
$ oc port-forward <collector_pod>
```

File: modules/distr-tracing-otel-troubleshoot-metrics.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-metrics.adoc:15]:
```suggestion
The operator creates a `<cr_name>-collector-monitoring` telemetry service that you can use to scrape the metrics endpoint.
```

File: modules/distr-tracing-otel-troubleshoot-logs.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-logs.adoc:24]: Please refer to the comment above.

**Review by michaelryanpeter — COMMENTED** [review]
I left some comments and suggestions to consider in a follow up issue or PR. 

They are all minor issues, so I do not want to block a merge over them. I created [TRACING-3591](https://issues.redhat.com/browse/TRACING-3591) to track the updates and assigned the issue to @max-cx.

Thank you for opening the PR!

File: modules/distr-tracing-otel-troubleshoot-logs.adoc
Comment (michaelryanpeter) [inline modules/distr-tracing-otel-troubleshoot-logs.adoc:8]: I think this introduction could be a little more specific. Maybe something like, "You can specify your desired log level for the OpenTelemetry Collector as shown in the following example..."

**Comment by michaelryanpeter** [comment]
@max-cx It looks like 4.11 and 4.12 will need manual cherry picks. Would you be able to take care of this?

**Comment by max-cx** [comment]
@michaelryanpeter 

1. Thank you so much for your review. I created a Jira and added it to my epic for the next release, so I'll make sure to update the docs with your suggestions.
2. Regarding cherry-picking to 4.11 and 4.12, I am a newbie to the OpenShift docs, so I unfortunately don't have any merge rights yet.

**Comment by michaelryanpeter** [comment]
@max-cx: NP! There are instructions for creating[ manual cherry picks in the OpenShift Docs manual.](https://docs.google.com/document/u/2/d/e/2PACX-1vRLKWEMHQ3DZroxZKfTu3XcrSdREr6D3oSSayBanEprXhkA2Ciyr2SQuDTYI4aIKUiOPPIQMHgjHeh8/pub#h.x2d4tdt5adta) You have to scroll a little bit, but the instructions are very detailed and should walk you through it. If you run into issues, please reach out in the team Slack channel. 

Please make sure you prepend the heading of the PR with `[enterprise-4.*]` and reference the commit message and original PR at the top of comment zero. These points are in instructions but are easy to miss sometimes. They make the manual cherrypicks consistent with the cherrypick bot. They also make it easier for the merge review squad to recognize that these are manual CPs.

Once you have created the manual CPs, put the `/label merge-review-needed` on.

**Comment by max-cx** [comment]
@michaelryanpeter, thank you so much for the link to the docs manual.

Unfortunately, I don't have commit rights, so I can't push my cherry-picks into any branch.

Also, these cherry-picks are currently blocked by the parent PR (https://github.com/openshift/openshift-docs/pull/62410) having not been cherry-picked yet.

**Comment by max-cx** [comment]
The parent PR (https://github.com/openshift/openshift-docs/pull/62410) has since been cherry-picked.

This PR is ready for the cherry-picks.

**Pull request merged**
